tests/storage/fsp/test_fsp_sr.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+import os
+from lib.common import wait_for, vm_image
+from lib.vdi import VDI
+
+@pytest.mark.usefixtures("fsp_device_config")
+class TestFSPSRCreateDestroy:
+    """
+    Tests that do not use fixtures that setup the SR
+    because they precisely need to test SR creation and destruction.
+    """
+
+    def test_create_and_destroy_sr(self, host, fsp_device_config):
+        pathtodb = fsp_device_config['pathtodb']
+        host.mkdir('', pathtodb + '/links')
+        sr = host.sr_create('fsp', "fsp-local-SR-test", {'file-uri': pathtodb})
+        sr.destroy()
+
+@pytest.mark.usefixtures("fsp_sr", "fsp_device_config")
+class TestFSPVDI:
+    def test_create_and_destroy_VDI(self, host, fsp_sr, fsp_device_config):
+        pathtoshareddir = os.path.split(fsp_device_config['pathtoshareddir'])
+        linkname = 'vdifor' + pathtoshareddir[-1]
+        host.ln(fsp_device_config['pathtoshareddir'], fsp_device_config['pathtodb'] + '/links/' + linkname)
+        fsp_sr.scan()
+        vdi_uuid = fsp_sr.vdi_uuids(managed=True, label=linkname)
+        # fail if the vdi could not be created or it already exists
+        assert len(vdi_uuid) == 1
+        vdi = VDI(fsp_sr, vdi_uuid[0])
+        vdi.destroy()
+        host.rm('-rf', fsp_device_config['pathtodb'] + '/links/' + linkname)
```

Comment:
You reversed the order of parameters here (should be path then flags... If we keep the Host.rm method as currently implemented).